src/Common/src/Abstractions/Steeltoe.Common.Abstractions.csproj

```diff
@@ -11,5 +11,6 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="$(ExtensionsVersion)" />
+    <PackageReference Include="System.Text.Json" Version="6.0.4" />
```

Comment:
I'm not sure about bringing System.Text.Json and the Json attributes into this package... the abstraction is supposed to be transport agnostic. For example, would this affect potential to add support for using Spring Boot Admin over gRPC?

Reply:
Note that #873 introduces a centralized version placeholder, so please use `$(SystemVersion)` once it has been merged.

Reply:
> I'm not sure about bringing System.Text.Json and the Json attributes into this package... the abstraction is supposed to be transport agnostic. For example, would this affect potential to add support for using Spring Boot Admin over gRPC?



> I'm not sure about bringing System.Text.Json and the Json attributes into this package... the abstraction is supposed to be transport agnostic. For example, would this affect potential to add support for using Spring Boot Admin over gRPC?

In 4.0, this is going to have to be completely redesigned. By exposing types that library users will need to implement health checks (ex HealthCheckResult) and statically typing it to the response of HealthEndpoint, AND using the same object for serialization, we are exposing too much to safely make changes in a backwards compatible way. It is hard to make a change without a breaking change. Considering the situation we are in my suggestion is take this dependency for now. When we do implement gRPC IMO the best thing would be to have separate packages for gRPC & Http.